sdk/identity/Azure.Identity/src/AuthorizationCodeCredential.cs

```diff
@@ -101,7 +101,7 @@ internal AuthorizationCodeCredential(string tenantId, string clientId, string cl
                           clientSecret,
                           options as ITokenCacheOptions,
                           null,
-                          options?.IsLoggingPIIEnabled ?? false);
+                          options?.Diagnostics?.IsExtendedUnsafeLoggingEnabled ?? false);
```

Comment:
nit: I think Diagnostics is always going to be non-null.

Reply:
Actually it could be null if someone derived from `TokenCredentialOptions` and set it to null. Perhaps I should make `TokenCredentialOptions` sealed to prevent this. Thoughts?

Reply:
I think leaving the null check is fine in that case.